src/plugin.js

```diff
@@ -107,7 +107,9 @@ const initPlugin = function(player, options) {
     // Disable timeouts in the background altogether according to the backgroundTimeout
     // option, or if the player is muted, as browsers may throttle javascript timers to
     // 1 minute in that case
-    if (document.visibilityState === 'hidden' && (options.backgroundTimeout === Infinity || player.muted())) {
+    if ((document.visibilityState === 'hidden' &&
+          (options.backgroundTimeout === Infinity || player.muted())) ||
+        options.timeout === Infinity) {
       return;
```

Comment:
Just to clarify the intent: this will disable all timeouts in both the foreground and background if `options.timeout === Infinity`?

Reply:
hm... good question. I think I had intended for each to be separate, but does that even make sense? I doubt anyone would not want a foreground timeout and want a background timeout.

Reply:
Yeah I really can't see any reason to disable foreground timeouts but leave background timeouts enabled. But on the flip side, having the foreground/background timeout settings completely independent is a more simple and explicit way to handle things.

Reply:
While it's a bit weird, I think keeping them separate makes more sense for this PR. So, going to add a check for document.visibilityState === visible in here.